Register Annotation resource.kubernetes.io/pod-claim-name (#47093)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

Note the [PodIndexLabel](/docs/reference/command-line-tools-reference/feature-gates/)
feature gate must be enabled for this label to be added to pods.

### resource.kubernetes.io/pod-claim-name

Type: Annotation

Example: `resource.kubernetes.io/pod-claim-name: "my-pod-claim"`

> Used on: Pod

Reviewer: ```suggestion
Used on: ResourceClaim
```

This annotation is assigned to generated ResourceClaims. 

> Its value corresponds to the `pod.spec.resourceClaims[].name` for which the ResourceClaim was created. 

Reviewer: ```suggestion
Its value corresponds to the name of the resource claim in the `.spec` of any Pod(s) for which
the ResourceClaim was created.
```

Author: Done!!

> This annotation is utilized internally by the Kubernetes controller.

Reviewer: ```suggestion
This annotation is an internal implementation detail of
[dynamic resource allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/).
You should not need to read or modify the value of this annotation.
```

Note the [PodIndexLabel](/docs/reference/command-line-tools-reference/feature-gates/)
feature gate must be enabled for this label to be added to pods.

### resource.kubernetes.io/pod-claim-name

Type: Annotation

Example: `resource.kubernetes.io/pod-claim-name: "my-pod-claim"`

Used on: ResourceClaim

This annotation is assigned to generated ResourceClaims. 
Its value corresponds to the name of the resource claim in the `.spec` of any Pod(s) for which the ResourceClaim was created.
This annotation is an internal implementation detail of [dynamic resource allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/).
You should not need to read or modify the value of this annotation.

> 

Reviewer 2: It's mentioned [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/resourceclaim/controller.go#L58-L59) that this annotation `is used only inside the controller and not documented as part of the Kubernetes API.`
It looks like it's intentionally not documented. @pohly is it?

Reviewer 3: This is indeed intentional.

Reviewer: OK, **but** per https://github.com/kubernetes/website/issues/47044 we don't have a way to register annotations without also documenting them.

Reviewer: (adding such a mechanism is welcome, and it's not actually a huge amount of work, but it'd be bigger than this PR)

Reviewer 2: /lgtm
